scene/resources/material.h

```diff
@@ -81,8 +81,12 @@ class Material : public Resource {
 class ShaderMaterial : public Material {
 	GDCLASS(ShaderMaterial, Material);
 	Ref<Shader> shader;
+	Ref<Shader> shader_temp;
```

Comment:
Not sure if the name is right, it's not temporary because it gets used onwards if the material has overrides. Maybe `shader_with_macro_override` would be more explicit?